realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -221,20 +221,44 @@ Table getTable(String className) {
     Table getTable(Class<? extends RealmModel> clazz) {
         Table table = classToTable.get(clazz);
         if (table == null) {
-            clazz = Util.getOriginalModelClass(clazz);
-            table = transaction.getTable(realm.configuration.getSchemaMediator().getTableName(clazz));
-            classToTable.put(clazz, table);
+            Class<? extends RealmModel> originalClass = Util.getOriginalModelClass(clazz);
+            if (originalClass != clazz) {
+                table = classToTable.get(originalClass);
+            }
+            if (table != null) {
+                // table was not cached against proxy class. Add cache entry.
```

Comment:
no I think.

If the table wasn't cached at all, `table` is always `null`.
Only when the `clazz` represents proxy class and the cache has and entry only for original class, we can reach here.